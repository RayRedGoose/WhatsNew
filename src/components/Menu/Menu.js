import React, { Component } from 'react';
import './Menu.scss';
import MenuItem from '../MenuItem/MenuItem';
import LocalImage from '../../images/placeholder.svg';
import TechImage from '../../images/technology.svg';
import EntrtImage from '../../images/tickets.svg';
import ScienceImage from '../../images/rocket.svg';
import HealthImage from '../../images/health.svg';
import PropTypes from 'prop-types';

class Menu extends Component {
  constructor(props) {
    super();
    this.state = {
      menuItems: [
        {id: 1, name:"local", icon:LocalImage, text: "Local news", isActive: true},
        {id: 2, name:"technology", icon:TechImage, text: "Technology", isActive: false},
        {id: 3, name:"entertainment", icon:EntrtImage, text: "Entertainment", isActive: false},
        {id: 4, name:"science", icon:ScienceImage, text: "Science", isActive: false},
        {id: 5, name:"health", icon:HealthImage, text: "Health", isActive: false}
      ]
    }
  }

  changeActiveTab = (activeItemName) => {
    const newStates = this.state.menuItems.map(item => {
      if (item.name === activeItemName) {
        item.isActive = true
      } else {
        item.isActive = false
      }
      return item
    })

    this.setState({menuItems: [...newStates]})
  }

  render() {
    const menuItemsElements = this.state.menuItems.map(item => {
      return (
        <MenuItem
          key = {item.id}
          elementClass={(item.isActive) ? "menu_item active-tab" : "menu_item"}
          name={item.name}
          iconName={item.icon}
          text={item.text}
          activateItem = {this.changeActiveTab}
          changeContent = {this.props.changeArticleTheme}
        />
      )
    })

    return (
      <aside className="menu">
        <h1>What's</h1>
        <nav>
          {menuItemsElements}
        </nav>
      </aside>
    );
  }
}

Menu.propTypes = {
  changeArticleTheme: PropTypes.func
}


export default Menu;
