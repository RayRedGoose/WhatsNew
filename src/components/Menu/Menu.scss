.menu {
  grid-area: menu;
  padding: 20px;
  h1 {
    color: #fff;
    font-size: 32px;
    text-transform: capitalize;
    line-height: 1.4em;
    padding-bottom: 20px;
    margin: 0 auto 20px;
    &::after {
      content: "New?";
      color: #f0aa34;
    }
  }
}
