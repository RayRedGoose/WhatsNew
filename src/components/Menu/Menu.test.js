import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';

describe('Menu', () => {
  let container;

  beforeEach(() => {
    container = shallow(<Menu
      changeArticleTheme={jest.fn()}
    />);
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(container).toMatchSnapshot();
  });

  it('should update state with menu items when changeActiveTab is called', () => {
    // Expectation
    expect(container.state('menuItems')[0].isActive).toEqual(true);
    expect(container.state('menuItems')[1].isActive).toEqual(false);
    // Execution
    container.instance().changeActiveTab('technology');

    // Expectation
    expect(container.state('menuItems')[0].isActive).toEqual(false);
    expect(container.state('menuItems')[1].isActive).toEqual(true);
  });
})
