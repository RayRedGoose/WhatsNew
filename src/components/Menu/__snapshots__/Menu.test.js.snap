// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot with all data passed in correctly 1`] = `
<aside
  className="menu"
>
  <h1>
    What's
  </h1>
  <nav>
    <MenuItem
      activateItem={[Function]}
      changeContent={[MockFunction]}
      elementClass="menu_item active-tab"
      iconName="placeholder.svg"
      key="1"
      name="local"
      text="Local news"
    />
    <MenuItem
      activateItem={[Function]}
      changeContent={[MockFunction]}
      elementClass="menu_item"
      iconName="technology.svg"
      key="2"
      name="technology"
      text="Technology"
    />
    <MenuItem
      activateItem={[Function]}
      changeContent={[MockFunction]}
      elementClass="menu_item"
      iconName="tickets.svg"
      key="3"
      name="entertainment"
      text="Entertainment"
    />
    <MenuItem
      activateItem={[Function]}
      changeContent={[MockFunction]}
      elementClass="menu_item"
      iconName="rocket.svg"
      key="4"
      name="science"
      text="Science"
    />
    <MenuItem
      activateItem={[Function]}
      changeContent={[MockFunction]}
      elementClass="menu_item"
      iconName="health.svg"
      key="5"
      name="health"
      text="Health"
    />
  </nav>
</aside>
`;
